File: JS/PGS_가장 가까운 같은 글자.js
function solution(s) {
    var answer = [];
    
    const arr = Array.from({length: 26}, () => -1);
    
    
    for(let i = 0 ; i < s.length ; i ++){
        if(arr[(s[i].charCodeAt(0))-97] == -1){
            answer.push(-1)
            arr[(s[i].charCodeAt(0))-97] = 0
        }else{
            answer.push(arr[(s[i].charCodeAt(0))-97])
            arr[(s[i].charCodeAt(0))-97] = 0
        }
        for(let j = 0 ; j < 26 ; j ++){
            if(arr[j] > -1){
                arr[j] += 1
            }
        }
    }
    
    return answer;
}

File: Python/PGS_가장 가까운 같은 글자.py
def solution(s):
    answer = []
    
    arr = [-1 for i in range(26)]
    
    for i in s:
        if arr[ord(i)-97] == -1:
            answer.append(-1)
            arr[ord(i)-97] = 0
        else:
            answer.append(arr[ord(i)-97])
            arr[ord(i)-97] = 0
            
        for a in range(len(arr)):
            if arr[a] >= 0:
                arr[a] += 1
    
    return answer